liblustreapi/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+            noprecreate: state & sys::obd_statfs_state_OS_STATE_NOPRECREATE != 0,
+            enospc: state & sys::obd_statfs_state_OS_STATE_ENOSPC != 0,
+            enoino: state & sys::obd_statfs_state_OS_STATE_ENOINO != 0,
+        }
+    }
+}
+
+#[derive(Copy, Clone)]
+pub struct Statfs {
+    pub ostype: u64,
+    pub blocks: u64,
+    pub bfree: u64,
+    pub bavail: u64,
+    pub files: u64,
+    pub ffree: u64,
+    pub fsid: Fid,
+    pub bsize: u32,
+    pub namelen: u32,
+    pub maxbytes: u64,
+    pub state: StatfsState,
+    pub fprecreated: u32,
+}
+impl From<sys::obd_statfs> for Statfs {
+    fn from(statfs: sys::obd_statfs) -> Self {
+        Statfs {
+            ostype: statfs.os_type,
+            blocks: statfs.os_blocks,
+            bfree: statfs.os_bfree,
+            bavail: statfs.os_bavail,
+            files: statfs.os_files,
+            ffree: statfs.os_ffree,
+            fsid: Fid::from(statfs.os_fsid),
+            bsize: statfs.os_bsize,
+            namelen: statfs.os_namelen,
+            maxbytes: statfs.os_maxbytes,
+            state: StatfsState::from(statfs.os_state),
+            fprecreated: statfs.os_fprecreated,
+        }
+    }
+}
+
+pub fn fid2path(device: &str, fidstr: &str) -> Result<String, io::Error> {
+    if !fidstr.starts_with('[') {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            format!("FID is invalid format {}", fidstr),
+        ));
+    }
+
+    let mut buf = Vec::with_capacity(PATH_BYTES);
+
+    let devptr = device.as_ptr() as *const i8;
+    let fidptr = fidstr.as_ptr() as *const i8;
+
+    let ptr = buf.as_mut_ptr() as *mut i8;
+
+    let rc = unsafe {
+        let mut recno: i64 = 0;
+        let mut linkno: i32 = 0;
+        sys::llapi_fid2path(
+            devptr,
+            fidptr,
+            ptr,
+            buf.capacity() as i32,
+            &mut recno as *mut std::os::raw::c_longlong,
+            &mut linkno as *mut std::os::raw::c_int,
+        )
+    };
+    if rc != 0 {
+        return Err(io::Error::from_raw_os_error(rc));
+    }
+
+    unsafe {
+        let s = CStr::from_ptr(ptr);
+        let len = s.to_bytes().len();
+        buf.set_len(len);
+    };
+
+    let cstr = CString::new(buf);
+    match cstr {
+        Err(x) => Err(io::Error::new(io::ErrorKind::InvalidData, x)),
+        Ok(x) => x
+            .into_string()
+            .map_err(|x| io::Error::new(io::ErrorKind::InvalidData, x)),
+    }
+}
+
+// pub fn obd_statfs(filepath: &String) -> Result<Statfs, io::Error> {
+//     let mut path = filepath;
+//     sys::llapi_obd_statfs(path.as_ptr(),
+//     Ok(Statfs::from(statfs))
+// }
+
+pub fn rmfid(device: &str, fidlist: impl IntoIterator<Item = String>) -> Result<(), io::Error> {
+    use std::fs; // replace with sys::llapi_rmfid
+
+    for fidstr in fidlist {
+        let path = fid2path(device, &fidstr)?;
+        fs::remove_file(path)?;
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
If we have any sample data it might be good to put a few scenarios in here unless it's pretty basic.